src/Tasks/Microsoft.NET.Build.Tasks/FilterResolvedFiles.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.Globalization;
+using NuGet.ProjectModel;
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    /// <summary>
+    /// Resolves the assemblies to be published for a .NET app.
+    /// </summary>
+    public class FilterResolvedFiles : TaskBase
+    {
+        private readonly List<ITaskItem> _assembliesToPublish = new List<ITaskItem>();
+        private readonly List<ITaskItem> _packagesResolved = new List<ITaskItem>();
+
+        [Required]
+        public string AssetsFilePath { get; set; }
+
+        [Required]
+        public ITaskItem[] ResolvedFiles { get; set; }
+
+        [Required]
+        public ITaskItem[] PackagesToPrune { get; set; }
+
+        [Required]
+        public string TargetFramework { get; set; }
+
+        public string RuntimeIdentifier { get; set; }
+
+        public string PlatformLibraryName { get; set; }
+
+        public bool IsSelfContained { get; set; }
+
+        /// <summary>
+        /// All the assemblies to publish.
+        /// </summary>
+        [Output]
+        public ITaskItem[] AssembliesToPublish
+        {
+            get { return _assembliesToPublish.ToArray(); }
```

Comment:
We have been moving away from this pattern in this repo.  See [https://github.com/dotnet/sdk/pull/1245#discussion_r118145348](https://github.com/dotnet/sdk/pull/1245#discussion_r118145348)